src/frontend/staff-portal/src/app/components/ticket-info/ticket-info.component.ts

```diff
@@ -1,147 +1,575 @@
+    if (allParts.length > 1) section = allParts[1].substring(0, allParts[1].length - 1);
+    if (allParts.length > 2) subsection = allParts[2].substring(0, allParts[2].length - 1);
+    if (allParts.length > 3) paragraph = allParts[3].substring(0, allParts[3].length - 1);
+
+    return { fullSection: fullSection, section: section, subsection: subsection, paragraph: paragraph };
+  }
+
+  public constructViolationTicketCount(statuteString: string): ViolationTicketCount {
+
+    this.tempViolationTicketCount = { description: "", actRegulation: "", fullSection: "", section: "", subsection: "", paragraph: "" };
+
+    // look in list of statutes
+    let statute = this.lookupsService.statutes.filter(x => x.statuteString == statuteString) as StatuteView[];
+    if (statute && statute.length > 0) {
+      this.tempViolationTicketCount.description = statute[0].description;
+      this.tempViolationTicketCount.actRegulation = statute[0].act; // to do break down legal paragraphing
+      let parts = this.unLegalParagraph(statute[0].section);
+      this.tempViolationTicketCount.fullSection = parts.fullSection;
+      this.tempViolationTicketCount.section = parts.section;
+      this.tempViolationTicketCount.subsection = parts.subsection;
+      this.tempViolationTicketCount.paragraph = parts.paragraph;
+      return this.tempViolationTicketCount;
+    }
+
+    // if not found in list of statutes, make an attempt from passed in string
+    // first portion before space is act or regulation
+    // second portion before next space is legal paragraphing
+    // remaining is desciption
+    const statuteStringParts = statuteString.split(" ");
+
+    // three parts yay
+    if (statuteStringParts.length > 2) {
+      this.tempViolationTicketCount.actRegulation = statuteStringParts[0];
+      let parts = this.unLegalParagraph(statuteStringParts[1]);
+      this.tempViolationTicketCount.fullSection = parts.fullSection;
+      this.tempViolationTicketCount.section = parts.section;
+      this.tempViolationTicketCount.subsection = parts.subsection;
+      this.tempViolationTicketCount.paragraph = parts.paragraph;
+      this.tempViolationTicketCount.description = statuteString.substring(
+        statuteString.indexOf(statuteStringParts[2])
+      );
+    }
+
+    // two parts hmm....
+    else if (statuteStringParts.length > 1) {
+      this.tempViolationTicketCount.actRegulation = "";
+      this.tempViolationTicketCount.fullSection = statuteStringParts[0];
+      this.tempViolationTicketCount.description = statuteString.substring(statuteString.indexOf(statuteStringParts[1]));
+
+      // yuk shove it in description field
+    } else {
+      this.tempViolationTicketCount.actRegulation = "";
+      this.tempViolationTicketCount.fullSection = "";
+      this.tempViolationTicketCount.description = statuteString;
+    }
+
+    return this.tempViolationTicketCount;
+  }
+
+  public enableNoticeOfDisputeSave(): boolean {
+
```

Comment:
same as contact info

Reply:
i will go back and possibly change this for contact info but must be like thjis for ticket info